
```{r}
file_of_interest <- read.csv("new_data/data_hupAS_RNAseq.txt", sep='\t')
data_genome <- read.csv("genes_scoelicolor.txt", sep = '')

```

```{r}
merged_data <- file_of_interest %>%
  left_join(data_genome, by = "gene")
  
 write.csv(merged_data, "final_data/argR_2018.csv", row.names = FALSE)

```

```{r}
getwd()
```

```{r}

data_list <- unique(c(bajojajo$add_variable))
uq_data_list <- unique(data_list)

```



```{r}
 
M145vsMGM10 <- read.csv("raw_data/RNAseq_Martyna/M145vsMGM10.csv", sep = ";")
M145vsMGM11 <- read.csv("raw_data/RNAseq_Martyna/M145vsMGM11.csv", sep = ";")
M145vsMGM12 <- read.csv("raw_data/RNAseq_Martyna/M145vsMGM12.csv", sep = ";")
M145vsMGM14 <- read.csv("raw_data/RNAseq_Martyna/M145vsMGM14.csv", sep = "\t")
PS04vsMGHM5 <- read.csv("raw_data/RNAseq_Martyna/PS04vsMGHM5.csv", sep = ";")
PS04vsMGP10 <- read.csv("raw_data/RNAseq_Martyna/PS04vsMGP10.csv", sep = ";")
PS04vsMGP11 <- read.csv("raw_data/RNAseq_Martyna/PS04vsMGP11.csv", sep = ";")
PS04vsMGP12 <- read.csv("raw_data/RNAseq_Martyna/PS04vsMGP12.csv", sep = ";")
PS04vsMGP14 <- read.csv("raw_data/RNAseq_Martyna/PS04vsMGP14.csv", sep = ";")

 
M145vsMGM10$add_variable <- "M145vsMGM10"
M145vsMGM11$add_variable <- "M145vsMGM11"
M145vsMGM12$add_variable <- "M145vsMGM12"
M145vsMGM14$add_variable <- "M145vsMGM14"
PS04vsMGHM5$add_variable <- "PS04vsMGHM5"
PS04vsMGP10$add_variable <- "PS04vsMGP10"
PS04vsMGP11$add_variable <- "PS04vsMGP11"
PS04vsMGP12$add_variable <- "PS04vsMGP12"
PS04vsMGP14$add_variable <- "PS04vsMGP14"

PS04vsMGP14 <- PS04vsMGP14[, -c(9, 10)]


###M145vsMGM10, M145vsMGM11, M145vsMGM12, M145vsMGM14, PS04vsMGHM5, PS04vsMGP10, PS04vsMGP11, PS04vsMGP12, PS04vsMGP14




RNAseq_Martyna <- rbind(M145vsMGM10, M145vsMGM11, M145vsMGM12, M145vsMGM14, PS04vsMGHM5, PS04vsMGP10, PS04vsMGP11, PS04vsMGP12, PS04vsMGP14)


```

```{r}
names(M145vsMGM10)
names(M145vsMGM11)
names(M145vsMGM12)
names(M145vsMGM14)
names(PS04vsMGHM5)
names(PS04vsMGP10)
names(PS04vsMGP11)
names(PS04vsMGP12)
names(PS04vsMGP14)

colnames(M145vsMGM14)[11] <- "log2.fold."

```


```{r}
RNAseq_Martyna_merged <- RNAseq_Martyna %>%
  mutate(
    start = coalesce(Transcription.Start, Translation.Start),  # Take start1 if available, otherwise take start2
    end = coalesce(Translation.Stop, Transcription.Stop)         # Take end1 if available, otherwise take end2
  ) %>%
  select(-Transcription.Start, -Translation.Start, -Translation.Stop, -Transcription.Stop)

jajko <- list("gene", "logFC", "FDR", "add_variable", 'start', 'end', 'width', 'strand', 'data_name' )

##RNAseq_Martyna_merged_właściwe <- data.frame(jajko)
##colnames(RNAseq_Martyna_merged_właściwe) <- jajko

char_vector <- RNAseq_Martyna_merged$log2.fold.
RNAseq_Martyna_merged$log2.fold. <- gsub(",", ".", RNAseq_Martyna_merged$log2.fold.)
RNAseq_Martyna_merged$qValue <- gsub(",", ".", RNAseq_Martyna_merged$qValue)

RNAseq_Martyna_merged$gene <- RNAseq_Martyna_merged$Synonym
RNAseq_Martyna_merged$logFC <- as.numeric(RNAseq_Martyna_merged$log2.fold.)
RNAseq_Martyna_merged$wchich_numeric <- non_numeric
RNAseq_Martyna_merged$FDR <- as.numeric(RNAseq_Martyna_merged$qValue)
RNAseq_Martyna_merged$add_variable <- RNAseq_Martyna_merged$add_variable
RNAseq_Martyna_merged$start <- RNAseq_Martyna_merged$start
RNAseq_Martyna_merged$end <- RNAseq_Martyna_merged$end
RNAseq_Martyna_merged$strand <- RNAseq_Martyna_merged$Strand
RNAseq_Martyna_merged$width <- RNAseq_Martyna_merged$fold
RNAseq_Martyna_merged$data_name <- "RNAseq_Martyna"
RNAseq_Martyna_merged_właściwe1 <- RNAseq_Martyna_merged %>% select("gene", "logFC", "FDR", "add_variable", 'start', 'end', 'width', 'strand', 'data_name' )

RNAseq_Martyna <- read.csv("final_data/RNAseq_Martyna.csv", sep = ";")
RNAseq_Martyna$logFC <- as.numeric(RNAseq_Martyna$logFC)

typeof(RNAseq_Martyna_merged$log2.fold.)
RNAseq_Martyna_merged_właściwe1$strand <- gsub("+", "1", RNAseq_Martyna_merged_właściwe1$strand)
RNAseq_Martyna_merged_właściwe1$strand <- gsub("-", "0", RNAseq_Martyna_merged_właściwe1$strand)


write.csv(RNAseq_Martyna_merged_właściwe1, "final_data/RNAseq_Martyna.csv", row.names = FALSE)



RNAseq_Martyna_merged_właściwe1 <- RNAseq_Martyna_merged_właściwe1 %>% mutate(strand_plot = ifelse(strand == '-', 0, 1))

RNAseq_Martyna_merged_właściwe1 <- RNAseq_Martyna_merged_właściwe1[, -c(10)]

```

AS Problemem jest to, że w RNAseq_Martyna_merged starty i końce genów nie zawsze zgadzają się z tymi, które są w genes_scoelicolor.txt.

Ten RNA-seq był robiony już jakiś czas temu, możliwe, że na innej wersji chromosomu Streptomyces. Myślę, że trzeba by to wszystko ujednolicić i zamienić wartości kolumn start, end, width i strand na pochodzące z genes_scoelicolor.txt

Twój kod powyżej nie do końca mi działa  -  nie powstaje RNAseq_Martyna, bo nie zgadzają się nazwy kolumn. Zamiast wszystko teraz zmieniać poprawię tylko plik wyjściowy, może to później ujednolicisz?

```{r}
library(dplyr)
dane_martyna <- read.csv('final_data/RNAseq_Martyna.csv')

dane_martyna %>% select(gene, logFC, FDR, add_variable, data_name) %>%
  left_join(data_genome) %>%
  select(gene, logFC, FDR, add_variable, start, end, width, strand, data_name) -> dane_martyna

write.csv(dane_martyna, "final_data/RNAseq_Martyna.csv", row.names = FALSE)

```

```{r}

M145PS04 <- read.csv("raw_data/szafran2019/M145PS04.txt", sep = "\t")
M145PS04novobiocyna <- read.csv("raw_data/szafran2019/M145PS04novobiocyna.txt", sep = "\t")
M145PS04thio <- read.csv("raw_data/szafran2019/M145PS04thio.txt", sep = "\t")
##processed_1M145_PS04Thio <- read.csv("raw_data/szafran2019/processed_1M145_PS04Thio.txt", sep = "\t")
##processed_1M145_PS04 <- read.csv("raw_data/szafran2019/processed_1M145_PS04.txt", sep = "\t")
##processed_1M145_M145Novob <- read.csv("raw_data/szafran2019/processed_1M145_M145Novob.txt", sep = "\t")

processed_1M145_PS04 <- read.csv("raw_data/szafran2019/processed_1M145_PS04.txt", 
                                sep = "\t",
                                fileEncoding = "UTF-16",
                                check.names = TRUE,
                                stringsAsFactors = FALSE)
processed_1M145_PS04Thio <- read.csv("raw_data/szafran2019/processed_1M145_PS04Thio.txt", 
                                sep = "\t",
                                fileEncoding = "UTF-16",
                                check.names = TRUE,
                                stringsAsFactors = FALSE)
processed_1M145_M145Novob <- read.csv("raw_data/szafran2019/processed_1M145_M145Novob.txt", 
                                sep = "\t",
                                fileEncoding = "UTF-16",
                                check.names = TRUE,
                                stringsAsFactors = FALSE)
```

```{r}

#mean_wildtype_M145PS04 <- rowMeans(cbind(M145PS04$Normalized.Counts.wildtype.Replicate.1, M145PS04$Normalized.Counts.wildtype.Replicate.2,
                             #   M145PS04$Normalized.Counts.wildtype.Replicate.3, M145PS04$Normalized.Counts.wildtype.Replicate.4))

#mean_novo10_M145PS04 <- rowMeans(cbind(M145PS04$Normalized.Counts.TopA_depletion.Replicate.1, M145PS04$Normalized.Counts.TopA_depletion.Replicate.2,
                               # M145PS04$Normalized.Counts.TopA_depletion.Replicate.3, M145PS04$Normalized.Counts.TopA_depletion.Replicate.4))

#mean_wildtype_M145PS04novobiocyna <- rowMeans(cbind(M145PS04novobiocyna$Normalized.Counts.wildtype.Replicate.1, M145PS04novobiocyna$Normalized.Counts.wildtype.Replicate.2,
                               # M145PS04novobiocyna$Normalized.Counts.wildtype.Replicate.3, M145PS04novobiocyna$Normalized.Counts.wildtype.Replicate.4))

#mean_novo10_M145PS04novobiocyna <- rowMeans(cbind(M145PS04novobiocyna$Normalized.Counts.novo10.Replicate.1, M145PS04novobiocyna$Normalized.Counts.novo10.Replicate.2))

#mean_wildtype_M145PS04thio <- rowMeans(cbind(M145PS04thio$Normalized.Counts.wildtype.Replicate.1, M145PS04thio$Normalized.Counts.wildtype.Replicate.2,
 #                               M145PS04thio$Normalized.Counts.wildtype.Replicate.3, M145PS04thio$Normalized.Counts.wildtype.Replicate.4))

#mean_novo10_M145PS04thio <- rowMeans(cbind(M145PS04thio$Normalized.Counts.TopA_overexpression.Replicate.1, #M145PS04thio$Normalized.Counts.TopA_overexpression.Replicate.2,
                              #  M145PS04thio$Normalized.Counts.TopA_overexpression.Replicate.3))

##mean_wildtype_processed_1M145_PS04 <- rowMeans(cbind(processed_1M145_PS04$Normalized.Counts.wildtype.Replicate.1, processed_1M145_PS04$Normalized.Counts.wildtype.Replicate.2,
                              #  processed_1M145_PS04$Normalized.Counts.wildtype.Replicate.3, processed_1M145_PS04$Normalized.Counts.wildtype.Replicate.4))

##mean_novo10_processed_1M145_PS04 <- rowMeans(cbind(processed_1M145_PS04$Normalized.Counts.TopA_depletion.Replicate.1, processed_1M145_PS04$Normalized.Counts.TopA_depletion.Replicate.2,
                               # processed_1M145_PS04$Normalized.Counts.TopA_depletion.Replicate.3, processed_1M145_PS04$Normalized.Counts.TopA_depletion.Replicate.4))

#mean_wildtype_processed_1M145_M145Novob <- rowMeans(cbind(processed_1M145_M145Novob$Normalized.Counts.wildtype.Replicate.1, processed_1M145_M145Novob$Normalized.Counts.wildtype.Replicate.2,
 #                               processed_1M145_M145Novob$Normalized.Counts.wildtype.Replicate.3, processed_1M145_M145Novob$Normalized.Counts.wildtype.Replicate.4))

#mean_novo10_processed_1M145_M145Novob <- rowMeans(cbind(processed_1M145_M145Novob$Normalized.Counts.novo10.Replicate.1, processed_1M145_M145Novob$Normalized.Counts.novo10.Replicate.2))

#mean_wildtype_processed_1M145_PS04Thio <- rowMeans(cbind(processed_1M145_PS04Thio$Normalized.Counts.wildtype.Replicate.1, processed_1M145_PS04Thio$Normalized.Counts.wildtype.Replicate.2,
 #                               processed_1M145_PS04Thio$Normalized.Counts.wildtype.Replicate.3, processed_1M145_PS04Thio$Normalized.Counts.wildtype.Replicate.4))

#mean_novo10_processed_1M145_PS04Thio <- rowMeans(cbind(processed_1M145_PS04Thio$Normalized.Counts.TopA_overexpression.Replicate.1, processed_1M145_PS04Thio$Normalized.Counts.TopA_overexpression.Replicate.2,
 #                               processed_1M145_PS04Thio$Normalized.Counts.TopA_overexpression.Replicate.3, processed_1M145_PS04Thio$Normalized.Counts.TopA_overexpression.Replicate.4))

```






```{r}


logFC_M145PS04 <- log2(mean_wildtype_M145PS04/mean_novo10_M145PS04)
logFC_M145PS04novobiocyna <- log2(mean_wildtype_M145PS04novobiocyna/mean_novo10_M145PS04novobiocyna)
logFC_M145PS04thio <- log2(mean_wildtype_M145PS04thio/mean_novo10_M145PS04thio)
logFC_processed_1M145_PS04Thio <- log2(mean_wildtype_processed_1M145_PS04Thio/mean_novo10_processed_1M145_PS04Thio)
logFC_processed_1M145_PS04 <- log2(mean_wildtype_processed_1M145_PS04/mean_novo10_processed_1M145_PS04)
logFC_processed_1M145_M145Novob <- log2(mean_wildtype_processed_1M145_M145Novob/mean_novo10_processed_1M145_M145Novob)

# AS zmieniłam żeby wartości zgadzały się z tymi opublikowanymi w Szafran 2019 et al

logFC_M145PS04 <- log2(M145PS04$Expression.TopA_depletion/M145PS04$Expression.wildtype)
logFC_M145PS04novobiocyna <- log2(M145PS04novobiocyna$Expression.novo10/M145PS04novobiocyna$Expression.wildtype)
logFC_M145PS04thio <- log2(M145PS04thio$Expression.TopA_overexpression/M145PS04thio$Expression.wildtype)


M145PS04$logFC <- logFC_M145PS04
M145PS04novobiocyna$logFC <- logFC_M145PS04novobiocyna
M145PS04thio$logFC <- logFC_M145PS04thio
# processed_1M145_PS04Thio$logFC <- logFC_processed_1M145_PS04Thio
# processed_1M145_PS04$logFC <- logFC_processed_1M145_PS04
# processed_1M145_M145Novob$logFC <- logFC_processed_1M145_M145Novob


M145PS04 <- M145PS04 %>% select(Synonym, logFC, qValue.wildtype.vs.TopA_depletion)
M145PS04novobiocyna <- M145PS04novobiocyna %>% select(Synonym, logFC, qValue.wildtype.vs.novo10)
M145PS04thio <- M145PS04thio %>% select(Synonym, logFC, qValue.wildtype.vs.TopA_overexpression)
# processed_1M145_PS04Thio <- processed_1M145_PS04Thio %>% select(Synonym, logFC, qValue.wildtype.vs.TopA_overexpression)
# processed_1M145_M145Novob <- processed_1M145_M145Novob %>% select(Synonym, logFC, qValue.wildtype.vs.novo10)
# processed_1M145_PS04 <- processed_1M145_PS04 %>% select(Synonym, logFC, qValue.wildtype.vs.TopA_depletion)

M145PS04
M145PS04novobiocyna
M145PS04thio
# processed_1M145_PS04Thio
# processed_1M145_M145Novob
# processed_1M145_PS04

# dane_martyna %>% select(gene, logFC, FDR, add_variable, data_name) %>%
#   left_join(data_genome) %>%
#   select(gene, logFC, FDR, add_variable, start, end, width, strand, data_name) -> dane_martyna

M145PS04$add_variable <- "M145PS04" 
M145PS04novobiocyna$add_variable <- "M145PS04novobiocyna" 
M145PS04thio$add_variable <- "M145PS04thio" 
# processed_1M145_PS04Thio$add_variable <- "processed_1M145_PS04Thio" 
# processed_1M145_M145Novob$add_variable <- "processed_1M145_M145Novob" 
# processed_1M145_PS04$add_variable <- "processed_1M145_PS04" 

M145PS04$data_name <- "szafran2019"
M145PS04novobiocyna$data_name <- "szafran2019"
M145PS04thio$data_name <- "szafran2019"
# processed_1M145_PS04Thio$data_name <- "szafran2019"
# processed_1M145_M145Novob$data_name <- "szafran2019"
# processed_1M145_PS04$data_name <- "szafran2019"

colnames(M145PS04) <- c("gene", "logFC", "FDR", "add_variable", "data_name")
colnames(M145PS04novobiocyna) <- c("gene", "logFC", "FDR", "add_variable", "data_name")
colnames(M145PS04thio) <- c("gene", "logFC", "FDR", "add_variable", "data_name")
# colnames(processed_1M145_PS04Thio) <- c("gene", "logFC", "FDR", "add_variable", "data_name")
# colnames(processed_1M145_M145Novob) <- c("gene", "logFC", "FDR", "add_variable", "data_name")
# colnames(processed_1M145_PS04) <- c("gene", "logFC", "FDR", "add_variable", "data_name")


# data_szafran2019 <- rbind(M145PS04, M145PS04novobiocyna, M145PS04thio, processed_1M145_PS04Thio, processed_1M145_M145Novob, processed_1M145_PS04)
data_szafran2019 <- rbind(M145PS04, M145PS04novobiocyna, M145PS04thio)


data_szafran2019 %>% select(gene, logFC, FDR, add_variable, data_name) %>%
  left_join(data_genome) %>%
  select(gene, logFC, FDR, add_variable, start, end, width, strand, data_name) -> data_szafran2019

data_szafran2019$FDR <- gsub(",", ".", data_szafran2019$FDR)
data_szafran2019$FDR <- as.numeric(data_szafran2019$FDR)
data_szafran2019 <- data_szafran2019 %>% mutate(strand_plot = ifelse(strand == '-', 0, 1))

summary(data_szafran2019)

data_szafran2019 <- data_szafran2019[, -c(10)]

write.csv(data_szafran2019, "final_data/data_szafran2019.csv", row.names = FALSE)

```


```{r}
jajo <- read.csv("usr_test.txt", sep = "\t")


jajo1 <- jajo %>% filter(logFC <= -1)



getwd()
```

```{r}

  filter_data_for_venn <- reactive({
    higher_logFC <- input$higher_logFC_venn
    lower_logFC <- input$lower_logFC_venn
    data_rna <- dataselection_venn()
    data_rna1 <- data_rna %>% filter(logFC >= higher_logFC, add_variable %in% c(input$contrast_venn_1, input$contrast_venn_2))
    data_rna2 <- data_rna %>% filter(logFC <= lower_logFC)
    data_rna3 <- data_rna %>% filter(logFC <= higher_logFC, add_variable %in% c(input$contrast_venn_1, input$contrast_venn_2))
    data_rna4 <- data_rna %>% filter(logFC >= lower_logFC)
    data_rna3$gene <- gsub("SC", "OD", data_rna3$gene)
    data_rna4$gene <- gsub("SC", "DO", data_rna4$gene)
    data_rna_filtered <- rbind(data_rna1, data_rna2, data_rna3, data_rna4)
    print(data_rna_filtered)
    if(switch_state_venn()) {
      data_rna_filtered1 <- data_rna_filtered %>% filter(FDR <= 0.05)
      data_rna_filtered2 <- data_rna_filtered %>% filter(FDR >= 0.05)
      data_rna_filtered2$gene <- gsub("SC", "JO", data_rna_filtered2$gene)
      data_rna_filtered <- rbind(data_rna_filtered1, data_rna_filtered2)
      return(data_rna_filtered)
    } else {
      return(data_rna_filtered)
    }
  })
  
```

```{r}

abrc3 <- read.csv("new_data/abrc3.txt", sep = "\t")
abrc3$logFC <- gsub("", "-", abrc3$logFC)
abrc3$logFC <- as.numeric(abrc3$logFC)
abrc3$add_variable <- abrc3$X 

abrc3 <- abrc3 %>% select(gene, FDR, logFC, add_variable)

merged_data <- abrc3 %>%
  left_join(data_genome, by = "gene")

```

```{r}

aor1_rna <- read.csv("new_data/aor1_rna.txt", sep = "\t")
aor1_rna$FC <- gsub(",", ".", aor1_rna$FC)
aor1_rna$FDR <- gsub(",", ".", aor1_rna$FDR)
aor1_rna$FC <- as.numeric(aor1_rna$FC)
aor1_rna$FDR <- as.numeric(aor1_rna$FDR)
aor1_rna$logFC <- log2(aor1_rna$FC)

aor1_rna <- aor1_rna %>% select(gene, FDR, logFC, add_variable)

df <- separate(aor1_rna, gene, into = c("gene", "number"), sep = "*", fill = "right")



merged_data <- aor1_rna %>%
  left_join(data_genome, by = "gene")

library(tidyr)



```

```{r}
argR_2018 <- read.csv("new_data/argR_2018.txt", sep = "\t")

argR_2018 <- argR_2018 %>% 
  pivot_longer(
    cols = !gene, 
    names_to = "add_variable", 
    values_to = "logFC"
  )

colnames(argR_2018)
argR_2018$"32h.DargR.M145" <- argR_2018$X32h.DargR.M145       
argR_2018$"42h.DargR.M145" <- argR_2018$X42h.DargR.M145       
argR_2018$'49h.DargR.M145' <- argR_2018$X49h.DargR.M145       
argR_2018$'56h.DargR.M145' <- argR_2018$X56h.DargR.M145       
argR_2018$'66h.DargR.M145' <- argR_2018$X66h.DargR.M145      
argR_2018$'32h.p.BH.DargR.M145' <- argR_2018$X32.h.p.BH.DargR.M145 
argR_2018$'42h.p.BH.DargR.M145' <- argR_2018$X42h.p.BH.DargR.M145  
argR_2018$'49h.p.BH.DargR.M145' <- argR_2018$X49h.p.BH.DargR.M145  
argR_2018$'56h.p.BH.DargR.M145' <- argR_2018$X56h.p.BH.DargR.M145  
argR_2018$'66h.p.BH.DargR.M145' <- argR_2018$X66h.p.BH.DargR.M145 

argR_2018 <- argR_2018 %>% select(gene, "32h.DargR.M145", "42h.DargR.M145", '49h.DargR.M145', '56h.DargR.M145', '66h.DargR.M145', '32h.p.BH.DargR.M145', '42h.p.BH.DargR.M145', '49h.p.BH.DargR.M145', '56h.p.BH.DargR.M145', '66h.p.BH.DargR.M145')


argR_2018$logFC <- argR_2018$logFC <- gsub(",", ".", argR_2018$logFC)
argR_2018$logFC <- as.numeric(argR_2018$logFC)

argR_2018$FDR <- 0


merged_data <- argR_2018 %>%
  left_join(data_genome, by = "gene")

```
```{r}
 genelist_sven <- read.csv("sven_genes_vnz.txt", sep = '')

genelist_sven <- genelist_sven %>% select(gene, start, end, strand)

```

```{r}
bldD_scoe <- read.csv("wip/bldD_scoe.txt", sep = "\t")

merged_data <- bldD_scoe %>%
  left_join(data_genome, by = "gene")
  
 write.csv(merged_data, "final_data/bldD_scoe.csv", row.names = FALSE)
################

data_bldC <- read.csv("wip/data_bldC.txt", sep = " ")

data_bldC <- data_bldC %>% select(gene, h10_logFC)
data_bldC$logFC <- data_bldC$h10_logFC
data_bldC <- data_bldC %>% select(gene, logFC)
data_bldC$FDR <- 0
data_bldC$add_variable <- "data_bldC"

merged_data <- data_bldC %>%
  left_join(genelist_sven, by = "gene")

merged_data <- na.omit(merged_data)
merged_data <- merged_data %>% select(gene, logFC, FDR, add_variable, start, end, strand)
 write.csv(merged_data, "final_data/data_bldC_sven.csv", row.names = FALSE)


```


```{r}

draRK_scoe <- read.csv("wip/draRK_scoe.txt", sep = "\t")

draRK_scoe <- draRK_scoe %>% select(gene, X36h, X48h, X60h)

draRK_scoe$DdraRKvsM145_36h <- draRK_scoe$X36h
draRK_scoe$DdraRKvsM145_48h <- draRK_scoe$X48h
draRK_scoe$DdraRKvsM145_60h <- draRK_scoe$X60h

draRK_scoe <- draRK_scoe %>% select(gene, DdraRKvsM145_36h, DdraRKvsM145_48h, DdraRKvsM145_60h)
draRK_scoe <- draRK_scoe %>% 
  pivot_longer(
    cols = !gene, 
    names_to = "add_variable", 
    values_to = "logFC"
  )

draRK_scoe$FDR <- 0

merged_data <- draRK_scoe %>%
  left_join(data_genome, by = "gene")
 write.csv(merged_data, "final_data/draRK_scoe.csv", row.names = FALSE)


```

```{r}

ECF42s_sven <- read.csv('wip/ECF42s_sven.txt', sep = '\t')

sven_convert <- read.csv('sven_genes_vnz_locus.txt', sep = ' ')
sven_convert$gene <- tolower(sven_convert$locus_sven)
merged_data <- ECF42s_sven %>%
  left_join(sven_convert, by = "gene")
merged_data$logFC <- log2(merged_data$FC)


merged_data <- merged_data %>% select(locus_vnz, logFC)
merged_data$gene <- merged_data$locus_vnz
merged_data <- merged_data %>% select(gene, logFC)
merged_data <- na.omit(merged_data)
merged_data <- merged_data %>%
  left_join(genelist_sven, by = "gene")
merged_data$add_variable <- "no_info"

 write.csv(merged_data, "final_data/ECF42s_sven.csv", row.names = FALSE)

```

```{r}
glmr_sven <- read.csv('wip/glmr_sven.txt', sep = '\t')
glmr_sven$DglnrvsM1246_30h <- glmr_sven$FC.in.WT.at.T30
glmr_sven$DglnrvsM1246_45h <- glmr_sven$FC.in.WT.at.T45

glmr_sven$DglnrvsM1246_30h <- log2(glmr_sven$DglnrvsM1246_30h) 
glmr_sven$DglnrvsM1246_45h <- log2(glmr_sven$DglnrvsM1246_45h)

glmr_sven_flip <- glmr_sven %>% select(gene, DglnrvsM1246_30h, DglnrvsM1246_45h)

glmr_sven_flip <- glmr_sven_flip %>% 
  pivot_longer(
    cols = !gene, 
    names_to = "add_variable", 
    values_to = "logFC"
  )

merged_data <- glmr_sven_flip %>%
  left_join(sven_convert, by = "gene")

merged_data$gene <- merged_data$locus_vnz

glmr_sven_flip <- merged_data %>% select(gene, logFC, add_variable)

glmr_sven_flip <- na.omit(glmr_sven_flip)

merged_data <- glmr_sven_flip %>%
  left_join(genelist_sven, by = "gene")

merged_data$FDR <- 0

 write.csv(merged_data, "final_data/glnr_sven.csv", row.names = FALSE)


DglnrvsM1246_30h
DglnrvsM1246_45h
M1246
```

```{r}

hups_rnaseq_Strzalka <- read.csv('wip/hups_rnaseq_Strzalka.txt', sep= '\t')

fdr_hups <- hups_rnaseq_Strzalka %>% select(gene,"hupS13.hupS22_fdr", "hupS13.wt13_fdr", "hupS22.wt22_fdr", "wt13.wt22_fdr")
FC_hups <- hups_rnaseq_Strzalka %>% select(gene, "hupS13.hupS22", "hupS13.wt13", "hupS22.wt22", "wt13.wt22")

fdr_hups <- fdr_hups %>% 
  pivot_longer(
    cols = !gene, 
    names_to = "add_variable_FDR", 
    values_to = "FDR"
  )



FC_hups <- FC_hups %>% 
  pivot_longer(
    cols = !gene, 
    names_to = "add_variable", 
    values_to = "logFC"
  )

FC_hups <- FC_hups %>% 
  replace_na(list(logFC = 0))

merged_data <- fdr_hups %>%
  left_join(FC_hups, by = "gene")



filtered_df <- merged_data %>%
  # Remove 'good_' prefix from the third column
  mutate(add_variable_FDR = sub("_fdr", "", add_variable_FDR)) %>%
  # Keep only rows where colors match
  filter(add_variable_FDR == add_variable)

filtered_df <- filtered_df %>% select(gene, logFC, FDR, add_variable)

merged_data <- filtered_df %>%
  left_join(genelist_sven, by = "gene")

 write.csv(merged_data, "final_data/hups_rnaseq_Strzalka.csv", row.names = FALSE)


```

```{r}

ohkA_scoe <- read.csv('wip/ohkA_scoe.txt', sep = '\t')

ohkA_scoe$DohkAvsM145_48h <- gsub(",", ".", ohkA_scoe$X48h_logFC)
ohkA_scoe$DohkAvsM145_64h <- gsub(",", ".", ohkA_scoe$X64h_logFC)
ohkA_scoe$DohkAvsM145_88h <- gsub(",", ".", ohkA_scoe$X88h_logFC)
ohkA_scoe$DohkAvsM145_48h <- as.numeric(ohkA_scoe$DohkAvsM145_48h)
ohkA_scoe$DohkAvsM145_64h <- as.numeric(ohkA_scoe$DohkAvsM145_64h)
ohkA_scoe$DohkAvsM145_88h <- as.numeric(ohkA_scoe$DohkAvsM145_88h)

ohkA_scoe <- ohkA_scoe %>% select(gene,DohkAvsM145_48h, DohkAvsM145_64h, DohkAvsM145_88h)

ohkA_scoe <- ohkA_scoe %>% 
  pivot_longer(
    cols = !gene, 
    names_to = "add_variable", 
    values_to = "logFC"
  )

ohkA_scoe$FDR <- 0

merged_data <- ohkA_scoe %>% left_join(data_genome, by = 'gene')

 write.csv(merged_data, "final_data/ohkA_scoe.csv", row.names = FALSE)


DohkAvsM145

```

```{r}
osdR_2016 <- read.csv('wip/osdR_2016.txt', sep = '\t')

osdR_2016 <- osdR_2016 %>% select("gene", "logFC_24h", "logFC_30h", "logFC_36h", "logFC_42h", "logFC_54h", "FDR")


osdR_2016_logFC <- osdR_2016 %>% select("gene", "DosdRvsM512_24h", "DosdRvsM512_30h", "DosdRvsM512_42h", "DosdRvsM512_36h", "DosdRvsM512_54h")
osdR_2016_FDR <- osdR_2016 %>% select("gene", "FDR")


osdR_2016_logFC <- osdR_2016_logFC %>% 
  pivot_longer(
    cols = !gene, 
    names_to = "add_variable", 
    values_to = "logFC"
  )


osdR_2016_FDR <- osdR_2016_FDR %>% 
  pivot_longer(
    cols = !gene, 
    names_to = "add_variable", 
    values_to = "FDR"
  )

osdR_2016_FDR <-  osdR_2016_FDR %>% select(gene, FDR)

merged_data <- osdR_2016_logFC %>% left_join(osdR_2016_FDR, by = 'gene')

merged_data <- merged_data %>% filter(merged_data$FDR != 'FDR')

osdR_2016$DosdRvsM512_30h <- osdR_2016$logFC_30h
osdR_2016$DosdRvsM512_36h <- osdR_2016$logFC_36h
osdR_2016$DosdRvsM512_42h <- osdR_2016$logFC_42h
osdR_2016$DosdRvsM512_54h <- osdR_2016$logFC_54h
osdR_2016$DosdRvsM512_24h <- osdR_2016$logFC_24h

DosdRvsM512

merged_data <- merged_data %>% left_join(data_genome, by = 'gene')


 write.csv(merged_data, "final_data/osdR_2016.csv", row.names = FALSE)

```


```{r}


sigR <- read.csv('wip/sigR.txt', sep = '\t')

sigR <- sigR %>% select(gene, logFC_deltasigR)

sigR$logFC <- sigR$logFC_deltasigR
sigR$add_variabble <- 'DsigRvsWT'

sigR <- sigR %>% select(gene, logFC, add_variabble)

sigR$FDR <- 0

merged_data <- sigR %>% left_join(data_genome, by = 'gene')


 write.csv(merged_data, "final_data/sigR.csv", row.names = FALSE)

```

```{r}

soxr_genes <- read.csv('wip/soxr_genes.txt', sep = '\t')

soxr1 <- soxr_genes %>% select(gene, FC_decrease_in_soxR_mutant_versus_WT_day3)
soxr2 <- soxr_genes %>% select(gene, FC_increase_in_WT_day3_vs_day1)
soxr1_FDR <- soxr_genes %>% select(gene, FDR)
soxr2_FDR <- soxr_genes %>% select(gene, FDR.1)


soxr1 <- soxr1 %>% 
  pivot_longer(
    cols = !gene, 
    names_to = "add_variable", 
    values_to = "logFC"
  )

soxr2 <- soxr2 %>% 
  pivot_longer(
    cols = !gene, 
    names_to = "add_variable", 
    values_to = "logFC"
  )


soxr1_FDR <- soxr1_FDR[-nrow(soxr1_FDR), ]
soxr2_FDR <- soxr2_FDR[-nrow(soxr2_FDR), ]
soxr1$logFC <- gsub(",", ".", soxr1$logFC)
soxr2$logFC <- gsub("<", "", soxr2$logFC)
soxr2$logFC <- gsub(">", "", soxr2$logFC)
soxr1$logFC <- as.numeric(soxr1$logFC)
soxr2$logFC <- as.numeric(soxr2$logFC)


soxr1$logFC <- log2(soxr1$logFC)
soxr2$logFC <- log2(soxr2$logFC)


soxr2_FDR$FDR <- soxr2_FDR$FDR.1 
soxr2_FDR <- soxr2_FDR %>% select(gene, FDR)


combined_df <- rbind(soxr1, soxr2)
combined_fdr <- rbind(soxr1_FDR, soxr2_FDR)
combined_fdr$FDR <- gsub(",", ".", combined_fdr$FDR)
combined_fdr$FDR <- as.numeric(combined_fdr$FDR)



combined_fdr_vector <- c(combined_fdr$FDR)


new_df <- bind_cols(combined_df, FDR = combined_fdr_vector)

new_df <- new_df %>% left_join(data_genome, by = 'gene')


 write.csv(new_df, "final_data/soxr_genes.csv", row.names = FALSE)

DsoxRvsWTday
```

```{r}

whiAH_scoe <- read.csv('wip/whiAH_scoe.txt', sep = '\t')
whiAH_scoe$whiAvsWT_18h <- whiAH_scoe$whiA18 
whiAH_scoe$whiAvsWT_36h <- whiAH_scoe$whiA36
whiAH_scoe$whiAvsWT_48h <- whiAH_scoe$whiA48
whiAH_scoe$whiHvsWT_18h <- whiAH_scoe$whiH18
whiAH_scoe$whiHvsWT_36h <- whiAH_scoe$whiH36
whiAH_scoe$whiHvsWT_48h <- whiAH_scoe$whiH48

whiAH_scoe <-  whiAH_scoe %>% select(gene,whiAvsWT_18h,whiAvsWT_36h,whiAvsWT_48h,whiHvsWT_18h,whiHvsWT_36h,whiHvsWT_48h)


whiAH_scoe$logFC <- gsub(",", ".", whiAH_scoe$logFC)
whiAH_scoe$logFC <- as.numeric(whiAH_scoe$logFC)

whiAH_scoe <- whiAH_scoe %>% 
  pivot_longer(
    cols = !gene, 
    names_to = "add_variable", 
    values_to = "logFC"
  )

whiAH_scoe <- na.omit(whiAH_scoe)

whiAH_scoe$FDR <- 0
whiAH_scoe <- whiAH_scoe %>% left_join(data_genome, by = 'gene')


 write.csv(whiAH_scoe, "final_data/whiAH_scoe.csv", row.names = FALSE)

```

```{r}
bldD_scoe <- read.csv("final_data/bldD_scoe.csv")
bldD_scoe$logFC <- gsub(",", ".", bldD_scoe$logFC)
bldD_scoe$logFC <- as.numeric(bldD_scoe$logFC)

 write.csv(bldD_scoe, "final_data/bldD_scoe.csv", row.names = FALSE)
```

```{r}
hups_rnaseq_Strzalka <- read.csv("final_data/hups_rnaseq_Strzalka.csv")

hups_rnaseq_Strzalka <- hups_rnaseq_Strzalka %>% select(gene, logFC, FDR, add_variable, start, end, strand)

   write.csv(hups_rnaseq_Strzalka, "final_data/hups_rnaseq_Strzalka_sven.csv", row.names = FALSE)

```

```{r}
osdr_20116 <- read.csv("final_data/osdr_2016.csv")

osdr_2016$logFC <- gsub(",", ".", osdr_2016$logFC)
osdr_2016$logFC <- as.numeric(osdr_2016$logFC)
osdr_2016$FDR <- gsub(",", ".", osdr_2016$FDR)
osdr_2016$FDR <- as.numeric(osdr_2016$FDR)

osdr_2016 <- na.omit(osdr_2016)

   write.csv(osdr_2016, "final_data/osdR_2016.csv", row.names = FALSE)


```

```{r}
sigR <- read.csv("final_data/sigR.csv")

sigR$add_variable <- sigR$add_variabble

sigR <- sigR %>% select(gene, logFC, FDR, add_variable, start, end, strand, width)
sigR$logFC <- gsub(",", ".", sigR$logFC)
sigR$logFC <- as.numeric(sigR$logFC)


   write.csv(sigR, "final_data/sigR.csv", row.names = FALSE)

```

```{r}

glnr_sven <- read.csv("final_data/glnr_sven.csv")

ECF42s_sven$FDR <- 0

glnr_sven <- glnr_sven %>% select(gene, logFC, add_variable, FDR, start, end, strand)

   write.csv(glnr_sven, "final_data/glnr_sven.csv", row.names = FALSE)



```

```{r}

abrB1.2_table4 <- read.csv("final_data/abrB1.2_table4.csv")
  abrB1.2_table4$add_variable <- "abrB1.2_table4"
  abrB1.2_table5 <- read.csv("final_data/abrB1.2_table5.csv")
  abrB1.2_table5$add_variable <- "abrB1.2_table5"
  

 abrB1.2_table <- rbind(abrB1.2_table4, abrB1.2_table5)
  abrB1.2_table$data_name <- "abrB1.2_table"


  abrc3 <- read.csv("final_data/abrc3.csv")
  abrc3$data_name <- "abrc3"


getwd()

data_meme <- rbind(abrc3, abrB1.2_table)

      data_meme1 <- data_meme %>% filter(FDR <= 0.05)

data_meme2 <- split(data_meme1$gene, data_meme1$data_name)

list1 <- data_meme2[[1]]
list2 <- data_meme2[[2]]

data_meme3 <- intersect(list1, list2)

filter(data_meme$gene, .by = data_meme3)

filtered <- data_meme[data_meme$gene == data_meme3 ]


filtered_df <- data_meme %>% filter(gene == data_meme3)

```
```{r}

yeong <- read.csv("new_data/yague_2013_scoe_diff.txt", sep = "\t")
yague <- read.csv("new_data/yeong_2016.txt",  sep = "\t")

colnames(yeong)
colnames(yague)


yeong$Average.log2.MII24h.MI12h  <- gsub(",", ".", yeong$Average.log2.MII24h.MI12h)
yeong$Average.log2.MII48h.MI12h  <- gsub(",", ".", yeong$Average.log2.MII48h.MI12h)
yeong$Average.log2.MII72h.MI12h  <- gsub(",", ".", yeong$Average.log2.MII72h.MI12h)
yeong$Average.log2.MII24h.MI12h <- as.numeric(yeong$Average.log2.MII24h.MI12h)
yeong$Average.log2.MII48h.MI12h <- as.numeric(yeong$Average.log2.MII48h.MI12h)
yeong$Average.log2.MII72h.MI12h <- as.numeric(yeong$Average.log2.MII72h.MI12h)

yeong$MII24hvsMI12h <- yeong$Average.log2.MII24h.MI12h
yeong$MII48hvsMI12h <- yeong$Average.log2.MII48h.MI12h
yeong$MII72hvsMI12h <- yeong$Average.log2.MII72h.MI12h
yeong$gene <- yeong$Gene

yeong <- yeong %>% select(gene, MII24hvsMI12h, MII48hvsMI12h, MII72hvsMI12h)

yeong <- yeong %>% 
  pivot_longer(
    cols = !gene, 
    names_to = "add_variable", 
    values_to = "logFC"
  )


yeong$FDR <- 0


yeong <- yeong %>%
  left_join(data_genome, by = "gene")
  
 write.csv(yeong, "final_data/yague_2013_scoe_diff.csv", row.names = FALSE)


getwd()

yague$MvsTFC  <- gsub(",", ".", yague$MvsTFC)
yague$MvsLFC  <- gsub(",", ".", yague$MvsLFC)
yague$MvsSFC  <- gsub(",", ".", yague$MvsSFC)
yague$TvsLFC  <- gsub(",", ".", yague$TvsLFC)
yague$TvsSFC  <- gsub(",", ".", yague$TvsSFC)
yague$LvsSFC  <- gsub(",", ".", yague$LvsSFC)
yague$MvsTFC <- as.numeric(yague$MvsTFC)
yague$MvsLFC <- as.numeric(yague$MvsLFC)
yague$MvsSFC <- as.numeric(yague$MvsSFC)
yague$TvsLFC <- as.numeric(yague$TvsLFC)
yague$TvsSFC <- as.numeric(yague$TvsSFC)
yague$LvsSFC <- as.numeric(yague$LvsSFC)

yague$MvsT <- yague$MvsTFC
yague$MvsL <- yague$MvsLFC
yague$MvsS <- yague$MvsSFC
yague$TvsL <- yague$TvsLFC
yague$TvsS <- yague$TvsSFC
yague$LvsS <- yague$LvsSFC

yague <- yague %>% select(gene, MvsT, MvsL, MvsS, TvsL, TvsS, LvsS)


yague <- yague %>% 
  pivot_longer(
    cols = !gene, 
    names_to = "add_variable", 
    values_to = "logFC"
  )

yague$logFC <- log2(yague$logFC)
yague$FDR <- 0

yague <- yague %>%
  left_join(data_genome, by = "gene")
  
 write.csv(yague, "final_data/yague_2016.csv", row.names = FALSE)

```
```{r}


  yague_2013_scoe_diff <- read.csv("final_data/yague_2013_scoe_diff.csv")
  yague_2013_scoe_diff$data_name <- "yague_2013_scoe_diff"
  whiAH_scoe <- read.csv("final_data/whiAH_scoe.csv")
  whiAH_scoe$data_name <- "whiAH_scoe"
  df <- data.frame(A1 = 1:3, B2 = 4:6, C3D = 7:9)
  
  
  combined_df <- rbind(whiAH_scoe, yague_2013_scoe_diff)
  
  yague_2013_scoe_diff$add_variable1 <- gsub("[^0-9]", "", yague_2013_scoe_diff$add_variable)
 
  yague_2013_scoe_diff <- yague_2013_scoe_diff[order(yague_2013_scoe_diff$add_variable1), ]
  
  filterowane <- yague_2013_scoe_diff %>% filter(gene %in% c("SCO0591", "SCO3311", "SCO2531"))
  filterowane1 <- whiAH_scoe %>% filter(gene %in% c("SCO0591", "SCO3311", "SCO2531"))
  
  
  filterowane2 <- rbind(filterowane, filterowane1)
  
  intersect(whiAH_scoe$gene, yague_2013_scoe_diff$gene)
  

   
   ggplot(filterowane2, aes(x = add_variable, y = logFC, 
                        color = gene, shape = data_name, group = interaction(gene, data_name))) +
  geom_point(size = 3) +  # Points for each condition
  geom_line() +           # Connecting lines
  geom_text(aes(label = gene), 
            data = filterowane2 %>% group_by(gene) %>% filter(logFC == max(logFC)), 
            hjust = -0.2, vjust = -0.5, size = 4) +  # Labeling at max logFC
  theme_minimal() +
  labs(title = "Gene Expression Changes from Two Datasets",
       x = "Condition",
       y = "Log Fold Change (logFC)",
       color = "Gene",
       shape = "Dataset") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
```
```{r}
abrc3 <- read.csv("final_data/abrc3.csv")
argR_2018 <- read.csv("final_data/argR_2018.csv")
draRK_scoe <- read.csv("final_data/draRK_scoe.csv")
ohkA_scoe <- read.csv("final_data/ohkA_scoe.csv")
osdr_2016 <- read.csv("final_data/osdr_2016.csv")
whiAH_scoe <- read.csv("final_data/whiAH_scoe.csv")
glnr_sven <- read.csv("final_data/glnr_sven.csv")
yague_2013_scoe_diff <- read.csv("final_data/yague_2013_scoe_diff.csv")





select_columns <- function(df, columns) {
  df_name <- deparse(substitute(df))
  selected_df <- df[, columns, drop = FALSE] 
  new_name <- paste0(df_name, "_selected") 
  assign(new_name, selected_df, envir = .GlobalEnv) 
  return(invisible(selected_df)) 
}


select_columns(yague_2013_scoe_diff, c("gene", "logFC", "FDR", "add_variable"))


add_data_name_column <- function(data_frames) {
  for (df_name in data_frames) {
    df <- get(df_name)  # Pobierz ramkę danych na podstawie nazwy
    
    # Usuń "_selected" z nazwy ramki
    original_name <- gsub("_selected$", "", df_name)
    
    # Dodaj kolumnę "data_name"
    df$data_name <- original_name
    
    # Nadpisz ramkę danych w globalnym środowisku
    assign(df_name, df, envir = .GlobalEnv)
  }
}


add_data_name_column(list("abrc3_selected", "argR_2018_selected", "draRK_scoe_selected", "glnr_sven_selected", "ohkA_scoe_selected", "osdr_2016_selected", "whiAH_scoe_selected", "yague_2013_scoe_diff_selected"))


#"","", ""



abrc3_selected$time <- gsub("[^0-9]", "", abrc3_selected$add_variable)
abrc3_selected$add_variable <- "one_condition"
abrc3_selected$data_name <- "abrc3"
abrc3_selected$time <- as.numeric(abrc3_selected$time)
write.csv(abrc3_selected, "final_data/data_intime/abrc3_intime.csv", row.names = FALSE)



argR_2018_selected <- argR_2018_selected %>%
  separate(add_variable, into = c("time", "add_variable"), sep = "h.")
argR_2018_selected$time <- as.numeric(argR_2018_selected$time)

write.csv(argR_2018_selected, "final_data/data_intime/argR_2018_intime.csv", row.names = FALSE)

draRK_scoe_selected <- draRK_scoe_selected %>%
  separate(add_variable, into = c("add_variable", "time"), sep = "_")
draRK_scoe_selected$time <- gsub("[^0-9]", "", draRK_scoe_selected$time)
draRK_scoe_selected$time <- as.numeric(draRK_scoe_selected$time)
write.csv(draRK_scoe_selected, "final_data/data_intime/draRK_scoe_intime.csv", row.names = FALSE)


ohkA_scoe_selected <- ohkA_scoe_selected %>%
  separate(add_variable, into = c("add_variable", "time"), sep = "_")
ohkA_scoe_selected$time <- gsub("[^0-9]", "", ohkA_scoe_selected$time)
ohkA_scoe_selected$time <- as.numeric(ohkA_scoe_selected$time)
write.csv(ohkA_scoe_selected, "final_data/data_intime/ohkA_scoe_intime.csv", row.names = FALSE)

osdr_2016_selected <- osdr_2016_selected %>%
  separate(add_variable, into = c("add_variable", "time"), sep = "_")
osdr_2016_selected$time <- gsub("[^0-9]", "", osdr_2016_selected$time)
osdr_2016_selected$time <- as.numeric(osdr_2016_selected$time)
write.csv(osdr_2016_selected, "final_data/data_intime/osdr_2016_intime.csv", row.names = FALSE)

whiAH_scoe_selected <- whiAH_scoe_selected %>%
  separate(add_variable, into = c("add_variable", "time"), sep = "_")
whiAH_scoe_selected$time <- gsub("[^0-9]", "", whiAH_scoe_selected$time)
whiAH_scoe_selected$time <- as.numeric(whiAH_scoe_selected$time)
write.csv(whiAH_scoe_selected, "final_data/data_intime/whiAH_scoe_intime.csv", row.names = FALSE)

table(yague_2013_scoe_diff_selected$add_variable)



yague_2013_scoe_diff_selected <- yague_2013_scoe_diff_selected %>%
  mutate(
    time = str_extract(add_variable, "\\d+"),  # Extract numbers (24, 48, 72)
    add_variable = str_replace(add_variable, "\\d+h", "inT")  # Replace number+h with "inT"
  )
yague_2013_scoe_diff_selected$time <- as.numeric(yague_2013_scoe_diff_selected$time)

write.csv(yague_2013_scoe_diff_selected, "final_data/data_intime/yague_2013_scoe_diff_intime.csv", row.names = FALSE)


glnr_sven_selected <- glnr_sven_selected %>%
  separate(add_variable, into = c("add_variable", "time"), sep = "_")
glnr_sven_selected$time <- gsub("[^0-9]", "", glnr_sven_selected$time)
glnr_sven_selected$time <- as.numeric(glnr_sven_selected$time)
write.csv(whiAH_scoe_selected, "final_data/data_intime/glnr_sven_intime.csv", row.names = FALSE)



filterowane <- yague_2013_scoe_diff_selected %>% filter(gene %in% c("SCO0591", "SCO3311", "SCO2531"))
  filterowane1 <- whiAH_scoe_selected %>% filter(gene %in% c("SCO0591", "SCO3311", "SCO2531"))
  
  
  filterowane2 <- rbind(filterowane, filterowane1)
  
  
   ggplot(filterowane2, aes(x = time, y = logFC, 
                        color = gene, shape = data_name, group = interaction(gene, data_name))) +
  geom_point(size = 3) +  # Points for each condition
  geom_line() +           # Connecting lines
  geom_text(aes(label = gene), 
            data = filterowane2 %>% group_by(gene) %>% filter(logFC == max(logFC)), 
            hjust = -0.2, vjust = -0.5, size = 4) +  # Labeling at max logFC
  theme_minimal() +
  labs(title = "Gene Expression Changes from Two Datasets",
       x = "Condition",
       y = "Log Fold Change (logFC)",
       color = "Gene",
       shape = "Dataset") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

```
```{r}


#"SCO7412, SCO6205, SCO7237, SCO6516, SCO0666, SCO6706, SCO0897, SCO7290, SCO0722, SCO2633, SCO6744, SCO6408, SCO3281, SCO0896, SCO3073, SCO1868, SCO4803, SCO6513, SCO0201, SCO0213, SCO2353, SCO0834, SCO0219, SCO0200, SCO0180, SCO0217, SCO0208, SCO0210, SCO1294, SCO0919, SCO0199, SCO4982, SCO0212, SCO0216, SCO0204, SCO0218, SCO0215, SCO0173, SCO0198, SCO3467, SCO0220, SCO0247, SCO4684, SCO5921"


  data_hupAS_RNAseq <- read.csv("final_data/data_hupAS_RNAseq.csv")
  data_hupAS_RNAseq$data_name <- "data_hupAS_RNAseq"
  data_szafran2019 <- read.csv("final_data/data_szafran2019.csv")

  
  gene_string <- "SCO7412, SCO6205, SCO7237, SCO6516, SCO0666, SCO6706, SCO0897, SCO7290, SCO0722, SCO2633, SCO6744, SCO6408, SCO3281, SCO0896, SCO3073, SCO1868, SCO4803, SCO6513, SCO0201, SCO0213, SCO2353, SCO0834, SCO0219, SCO0200, SCO0180, SCO0217, SCO0208, SCO0210, SCO1294, SCO0919, SCO0199, SCO4982, SCO0212, SCO0216, SCO0204, SCO0218, SCO0215, SCO0173, SCO0198, SCO3467, SCO0220, SCO0247, SCO4684, SCO5921"


gene_vector <- unlist(strsplit(gene_string, ", "))

zbindowane <- rbind(data_hupAS_RNAseq, data_szafran2019)

filter_gene_vector <- zbindowane %>% filter(gene_vector)

filter_gene_vector <- zbindowane %>%
  filter(gene %in% gene_vector)


```

```{r}

jajko <- c("abrB1.2_table", "data_hupAS_RNAseq", "RNAseq_Martyna", "szafran2019", "abrc3", "aor1_rna","argR_2018", "bldD_scoe", "data_bldC_sven", "draRK_scoe", "ECF42s_sven", "glnr_sven", "hups_rnaseq_Strzalka_sven", "ohkA_scoe", "osdR_2016", "sigR", "soxr_genes", "whiAH_scoe", "yague_2013_scoe_diff", "yeong_2016")
sorted_list <- sort(jajko)

print(jajko)

#"abrB1.2_table"             "data_hupAS_RNAseq"         "RNAseq_Martyna"            "szafran2019"               "abrc3"                     "aor1_rna" "argR_2018"                 "bldD_scoe"                 "data_bldC_sven"            "draRK_scoe"                "ECF42s_sven"               "glnr_sven"                
#"hups_rnaseq_Strzalka_sven" "ohkA_scoe"                 "osdR_2016"                 "sigR"                      "soxr_genes"                "whiAH_scoe"               
#"yague_2013_scoe_diff"      "yeong_2016"     

#"abrB1.2_table", "data_hupAS_RNAseq", "RNAseq_Martyna", "szafran2019", "abrc3", "aor1_rna", "argR_2018", "bldD_scoe", "draRK_scoe", "ohkA_scoe", "osdR_2016", "sigR", "soxr_genes", "whiAH_scoe", "yague_2013_scoe_diff"      "yeong_2016", "data_bldC_sven", "ECF42s_sven", "glnr_sven", "hups_rnaseq_Strzalka_sven"

 

```
```{r}

sven_NRRL_metab_RNAseq <- read.csv("new_data/sven_NRRL_metab_RNAseq.txt",  sep = "\t")

sven_NRRL_metab_RNAseq <- sven_NRRL_metab_RNAseq %>% select(ID, log2_fc_18h, log2_fc_24h, log2_fc_48h)

sven_convert <- read.csv('sven_genes_vnz_locus.txt', sep = ' ')

sven_NRRL_metab_RNAseq$id <- sven_NRRL_metab_RNAseq$ID

merged_data <- sven_NRRL_metab_RNAseq %>%
  left_join(sven_convert, by = "id")

merged_data <- merged_data %>% select(locus_vnz, log2_fc_18h, log2_fc_24h, log2_fc_48h)

merged_data <- na.omit(merged_data)

merged_data$gene <- merged_data$locus_vnz

merged_data <- merged_data %>% select(gene, log2_fc_18h, log2_fc_24h, log2_fc_48h)

merged_data <- merged_data %>% 
  pivot_longer(
    cols = !gene, 
    names_to = "add_variable", 
    values_to = "logFC"
  )

merged_data$logFC <- gsub(",", ".", merged_data$logFC)
merged_data$logFC <- as.numeric(merged_data$logFC)

merged_data <- merged_data %>%
  separate(add_variable, into = c("time", "add_variable"), sep = "2_fc_")

merged_data$FDR <- 0
merged_data <- merged_data %>% select(gene, logFC, FDR, add_variable)



merged_data <- merged_data %>%
  left_join(genelist_sven, by = "gene")

merged_data <- merged_data %>% select(gene, logFC, FDR, add_variable, start, end, strand)
 write.csv(merged_data, "final_data/NRRL_metab_RNAseq_sven.csv", row.names = FALSE)

merged_data <- merged_data %>% select(gene, logFC, FDR, add_variable)
merged_data$data_name <- "NRRL_metab_RNAseq_sven"

merged_data <- merged_data %>%
  separate(add_variable, into = c("time", "add_variable"), sep = "h")

merged_data$time <- as.numeric(merged_data$time)

merged_data$add_variable <- "one condition"
 write.csv(merged_data, "final_data/data_intime/NRRL_metab_RNAseq_sven_intime.csv", row.names = FALSE)

```
```{r}
getwd()


  abrB1.2_table4 <- read.csv("final_data/abrB1.2_table4.csv")
  abrB1.2_table4$add_variable <- "DabrBvsWT24h"
  
  abrB1.2_table4$data_name <- "abrB1.2_table4"
  abrB1.2_table5 <- read.csv("final_data/abrB1.2_table5.csv")
  abrB1.2_table5$add_variable <- "DabrBvsWT36h"
  abrB1.2_table5$data_name <- "abrB1.2_table5"
  
  abr <- rbind(abrB1.2_table4, abrB1.2_table5)

  abr$data_name <- "AbrC3_rico_2014"
   write.csv(abr, "final_data/abrB1.2_table.csv", row.names = FALSE)
  abrB1.2_table <- read.csv("final_data/abrB1.2_table.csv")

 
```

```{r}
aor <- read.csv("final_data/aor1_rna.csv")
aor$add_variable <- "Daor1vsWT"

   write.csv(aor, "final_data/aor1_rna.csv", row.names = FALSE)


```

```{r}

agR <- read.csv("new_data/argR_2018.txt", sep = "\t")

agrR_logFC <- agR[, 1:6]
agrR_FDR <- agR[, c(1, 7:11)]
nazwy_kolumn <- c("gene", "DargRvsWT32h", "DargRvsWT42h", "DargRvsWT49h", "DargRvsWT56h", "DargRvsWT66h")

colnames(agrR_FDR) <- nazwy_kolumn
colnames(agrR_logFC) <- nazwy_kolumn


agrR_FDR <- agrR_FDR %>% 
  pivot_longer(
    cols = !gene, 
    names_to = "add_variable", 
    values_to = "FDR"
  )

agrR_logFC <- agrR_logFC %>% 
  pivot_longer(
    cols = !gene, 
    names_to = "add_variable", 
    values_to = "logFC"
  )


agrR_FDR$FDR <- gsub(",", ".", agrR_FDR$FDR)
agrR_logFC$logFC <- gsub(",", ".", agrR_logFC$logFC)
agrR_FDR$FDR <- as.numeric(agrR_FDR$FDR)
agrR_logFC$logFC <- as.numeric(agrR_logFC$logFC)

merged_data <- agrR_logFC %>%
  left_join(agrR_FDR, by = c("gene", "add_variable"))


merged_data <- merged_data %>%
  left_join(data_genome, by = "gene")
  
 write.csv(merged_data, "final_data/argR_2018.csv", row.names = FALSE)

```

```{r}
bldd <- read.csv("final_data/bldD_scoe.csv")

bldd$add_variable <- "DbldDvsWT"
bldd$logFC <- log2(bldd$logFC)

table(bldd$logFC)

bldd1 <- na.omit(bldd)
 write.csv(bldd1, "final_data/bldD_scoe.csv", row.names = FALSE)


```
```{r}
bldc <- read.csv("new_data/data_bldC.txt", sep = " ")

bldc <- bldc[, c(1, 4:7)]

bldc_logFC <- bldc[, c(1, 2, 4)] 
bldc_FDR <- bldc[, c(1, 3, 5)]

bldc_nazwy <- c("gene", "DbldCvsWT10h", "DbldCcsWT14h")

colnames(bldc_logFC) <- bldc_nazwy
colnames(bldc_FDR) <- bldc_nazwy


bldc_FDR <- bldc_FDR %>% 
  pivot_longer(
    cols = !gene, 
    names_to = "add_variable", 
    values_to = "FDR"
  )

bldc_logFC <- bldc_logFC %>% 
  pivot_longer(
    cols = !gene, 
    names_to = "add_variable", 
    values_to = "logFC"
  )


merged_data <- bldc_logFC %>%
  left_join(bldc_FDR, by = c("gene", "add_variable"))

sven_genom <- read.csv("sven_genes_vnz.txt", sep = " ")

merged_data <- merged_data %>%
  left_join(sven_genom, by = "gene")

merged_data <- na.omit(merged_data)

merged_data <- merged_data %>% select(-product)

 write.csv(merged_data, "final_data/data_bldC_sven.csv", row.names = FALSE)


```

```{r}
ecf <- read.csv("new_data/ECF42s_sven.txt", sep = "\t")
ecf$add_variable <- "ECF42 sigma factor"
ecf <- ecf %>% rename(locus_sven = gene)


ecf <- ecf %>%
  left_join(sven_convert, by = "locus_sven")

ecf <- ecf %>% rename(gene = locus_vnz)
ecf <- ecf %>%
  left_join(genelist_sven, by = "gene")

ecf <- ecf %>% select("gene", "logFC", "FDR", "add_variable", "start", "end", "strand")
ecf <- na.omit(ecf)

ecf$logFC <- gsub(",", ".", ecf$logFC)
ecf$FDR <- gsub(",", ".", ecf$FDR)
ecf$logFC <- as.numeric(ecf$logFC)
ecf$FDR <- as.numeric(ecf$FDR)


 write.csv(ecf, "final_data/ECF42s_sven.csv", row.names = FALSE)

```

```{r}
glnr <- read.csv("final_data/glnr_sven.csv")

glnr$add_variable <- gsub("h", "min", glnr$add_variable)
glnr$add_variable <- gsub("M1246", "WT", glnr$add_variable)

 write.csv(glnr, "final_data/glnr_sven.csv", row.names = FALSE)

```

```{r}
ohka <- read.csv("final_data/ohkA_scoe.csv")


ohka$logFC <- log2(ohka$logFC)

ohka <- head(ohka, -3)

 write.csv(ohka, "final_data/ohkA_scoe.csv", row.names = FALSE)


```

```{r}
osdr <- read.csv("final_data/osdR_2016.csv")


osdr$logFC <- log2(osdr$logFC)

ohka <- head(ohka, -3)

 write.csv(osdr, "final_data/osdR_2016.csv", row.names = FALSE)


```

```{r}
yague <- read.csv("new_data/yague_2013_scoe_diff.txt", sep = "\t")

yague$logFC <- gsub(",", ".", yague$logFC)
yague$FDR <- gsub(",", ".", yague$FDR)
yague$logFC <- as.numeric(yague$logFC)
yague$FDR <- as.numeric(yague$FDR)

yague <- yague %>%
  left_join(data_genome, by = "gene")

write.csv(yague, "final_data/yague_2013_scoe_diff.csv", row.names = FALSE)

```

```{r}
yong <- read.csv("new_data/yeong_2016.txt", sep = "\t")
yong_logFC <- yong[, c(1, 2, 4, 6, 8, 10, 12)]
yong_FDR <- yong[, c(1, 3, 5, 7, 9, 11, 13)]

yong_nazwy <- c("gene", "MvsT", "MvsL", "MvsS", "TvsL", "TvsS", "LvsS")

colnames(yong_logFC) <- yong_nazwy
colnames(yong_FDR) <- yong_nazwy

yong_FDR <- yong_FDR %>% 
  pivot_longer(
    cols = !gene, 
    names_to = "add_variable", 
    values_to = "FDR"
  )

yong_logFC <- yong_logFC %>% 
  pivot_longer(
    cols = !gene, 
    names_to = "add_variable", 
    values_to = "logFC"
  )

yong <- yong_logFC %>%
  left_join(yong_FDR, by = c("gene", "add_variable"))

yong$logFC <- gsub(",", ".", yong$logFC)
yong$FDR <- gsub(",", ".", yong$FDR)
yong$logFC <- as.numeric(yong$logFC)
yong$FDR <- as.numeric(yong$FDR)

yong$logFC <- log2(yong$logFC)


yong <- yong %>%
  left_join(data_genome, by = "gene")

yong <- na.omit(yong)


write.csv(yong, "final_data/yeong_2016.csv", row.names = FALSE)


```

```{r}
edger <- read.csv("final_data/data_chip_seq/hupAS_strzalka/data_hupA_chipseq_edgeR.txt", sep = " ")
macs <- read.csv("final_data/data_chip_seq/hupAS_strzalka/data_hupA_chipseq_macs.txt", sep = " ")

edger <- edger %>% select(start, end, rodzaj)
macs <-  macs %>% select(start, end, rodzaj)

hupa <- rbind(edger, macs)

hupa$chrom <- "Chromosome"

hupa <- hupa[, c(4, 1, 2, 3)]

colnames(hupa) <- c("chrom", "chromStart", "chromEnd", "name")

write.csv(hupa, "final_data/data_chip_seq/hupAS_strzalka_chipseq.csv", row.names = FALSE)

```

```{r}
smirdova <- read.csv("final_data/data_chip_seq/chipseq_smirdova.txt", sep = "\t")
write.csv(smirdova, "final_data/data_chip_seq/chipseq_smirdova.csv", row.names = FALSE)

```

```{r}

SMC_14$name <- "SMC_14"
hupS_14$name <- "hupS_14"
dSMChupS_14$name <- "dSMChupS_14"
dparBSMC_14$name <- "dparBSMC_14"

hupS_chipseq_szafran <- rbind(SMC_14, hupS_14, dSMChupS_14, dparBSMC_14)

hupS_chipseq_szafran <- hupS_chipseq_szafran %>% select(chrom, chromStart, chromEnd, name) 

write.csv(hupS_chipseq_szafran, "final_data/data_chip_seq/hupS_chipseq_szafran.csv", row.names = FALSE)


```

€
```{r}


ggplot(hupS_chipseq_szafran, aes(y = name, x = start)) +
      geom_segment(aes(xend = end, yend = name, color = name),
                   size = 8, alpha = 0.6) +
      scale_color_brewer(palette = "Set2") +
      theme_minimal() +
      theme(
        legend.position = "none",
        panel.grid.minor = element_blank(),
        axis.title.y = element_blank(),
        plot.margin = margin(r = 100, l = 20, b = 20, t = 20)
      )


```



```{r}
baba <- c("coś", "łoś")
dziad <- c("niema", "jak")

mak <- list(baba, dziad)

mak
```
```{r}

yong <- read.csv("new_data/yeong_2016.txt", sep = "\t")
yong_logFC <- yong[, c(1, 2, 4, 6, 8, 10, 12)]
yong_FDR <- yong[, c(1, 3, 5, 7, 9, 11, 13)]

yong_nazwy <- c("gene", "MvsT", "MvsL", "MvsS", "TvsL", "TvsS", "LvsS")

colnames(yong_logFC) <- yong_nazwy
colnames(yong_FDR) <- yong_nazwy

yong_FDR <- yong_FDR %>% 
  pivot_longer(
    cols = !gene, 
    names_to = "add_variable", 
    values_to = "FDR"
  )

yong_logFC <- yong_logFC %>% 
  pivot_longer(
    cols = !gene, 
    names_to = "add_variable", 
    values_to = "logFC"
  )

yong <- yong_logFC %>%
  left_join(yong_FDR, by = c("gene", "add_variable"))

yong$logFC <- gsub(",", ".", yong$logFC)
yong$FDR <- gsub(",", ".", yong$FDR)
yong$logFC <- as.numeric(yong$logFC)
yong$FDR <- as.numeric(yong$FDR)

yong$logFC <- log2(yong$logFC)


yong <- yong %>%
  left_join(data_genome, by = "gene")

```

```{r}
library(stringr)

#data upload
data_test <- read.csv("test_no_format.txt", sep = '\t')

#spliting data and removing suffixes
dt_logFC <- data_test %>%
  select(gene, contains("logFC")) %>%
  rename_with(~str_remove(., "logFC"), -1)

dt_FDR <- data_test %>%
  select(gene, contains("FDR")) %>%
  rename_with(~str_remove(., "FDR"), -1)

#conversion to tidy
dt_logFC <- dt_logFC %>% 
  pivot_longer(
    cols = !gene, 
    names_to = "add_variable", 
    values_to = "logFC"
  )

dt_FDR <- dt_FDR %>% 
  pivot_longer(
    cols = !gene, 
    names_to = "add_variable", 
    values_to = "FDR"
  )

#conversion to numerical
dt_logFC$logFC <- gsub(",", ".", dt_logFC$logFC)
dt_FDR$FDR <- gsub(",", ".", dt_FDR$FDR)
dt_logFC$logFC <- as.numeric(dt_logFC$logFC)
dt_FDR$FDR <- as.numeric(dt_FDR$FDR)

#joining back into one
data_in_tidy <- dt_logFC %>%
  left_join(dt_FDR, by = c("gene", "add_variable"))

```

```{r}
adpa_rna <- read.csv("new_data/rnaseq_app_test.txt", sep='\t')
adpa_chip <- read.csv("new_data/adpa_chipseq.txt", sep='\t')


merged_data <- adpa_rna %>%
  left_join(genelist_sven, by = "gene")

merged_data <- na.omit(merged_data)
merged_data <- merged_data %>% select(gene, logFC, FDR, add_variable, start, end, strand)
 write.csv(adpa_chip, "final_data/data_chip_seq/adpa_wolanski_sven.csv", row.names = FALSE)

```

```{r}

# Sample text content (replace with your actual text)
text_content1 <- "
# Welcome to Page 1

This is the first text page. Replace this content with your own text.

## Section 1
Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.

## Section 2
Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
"

text_content2 <- "
# Welcome to Page 3

This is the third text page. Replace this content with your own text.

## Important Information
Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.

## Additional Notes
Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
"

ui <- page_navbar(
  title = "Multi-page App",
  
  nav_panel(
    "Text Page 1",
    card(
      card_header("Text Content"),
      card_body(
        markdown(text_content1)
      )
    )
  ),
  
  nav_panel(
    "Tables Page",
    layout_columns(
      card(
        card_header("Table 1"),
        card_body(
          DTOutput("table1")
        )
      ),
      card(
        card_header("Table 2"),
        card_body(
          DTOutput("table2")
        )
      )
    )
  ),
  
  nav_panel(
    "Text Page 2",
    card(
      card_header("More Text Content"),
      card_body(
        markdown(text_content2)
      )
    )
  ),
  
  theme = bs_theme(version = 5, bootswatch = "flatly")
)

server <- function(input, output, session) {
  # Render the first table
  output$table1 <- renderDT({
    datatable(table1, options = list(pageLength = 10))
  })
  
  # Render the second table
  output$table2 <- renderDT({
    datatable(table2, options = list(pageLength = 10))
  })
}

shinyApp(ui, server)


```

```{r}


data_template <- read.csv("templates_for_app.txt", sep='\t')
```

```{r}
MMC_rnaseq <- read.csv("new_data/MMC_rnaseq.txt", sep=' ')
bldB_rnaseq <- read.csv("new_data/bldB_rnaseq.txt", sep=' ')
ccr1_rnaseq <- read.csv("new_data/ccr1_rnaseq.txt", sep=' ')
ethanol_sekurowa_sven <- read.csv("new_data/ethanol.txt", sep=' ')
gbn_chipseq <- read.csv("new_data/gbn_chipseq.txt", sep='\t')
gbn_rnaseq <- read.csv("new_data/gbn_rnaseq.txt", sep=' ')


df <- read.csv("new_data/gbn_rnaseq.txt", sep = ' ', header = TRUE, stringsAsFactors = FALSE)

 write.csv(bldB_rnaseq, "final_data/BldB_Avramova_2023_sven.csv", row.names = FALSE)
 write.csv(MMC_rnaseq, "final_data/MMC_Stratton_2022_sven.csv", row.names = FALSE)
 write.csv(ccr1_rnaseq, "final_data/ccr1_Lei_2024.csv", row.names = FALSE)
 write.csv(ethanol_sekurowa_sven, "final_data/ETOHschock_Sekurova_2022_sven.csv", row.names = FALSE)
 write.csv(gbn_rnaseq, "final_data/Gbn_Du_2022.csv", row.names = FALSE)
 write.csv(gbn_chipseq, "final_data/data_chip_seq/Gbn_chipseq_Du_2022.csv", row.names = FALSE)



```

```{r}
MMC_rnaseq <- MMC_rnaseq %>%
  left_join(genelist_sven, by = "gene") 
bldB_rnaseq <- bldB_rnaseq %>%
  left_join(genelist_sven, by = "gene")
ccr1_rnaseq <- ccr1_rnaseq %>%
  left_join(data_genome, by = "gene")
ethanol_sekurowa_sven <- ethanol_sekurowa_sven %>%
  left_join(genelist_sven, by = "gene")
gbn_rnaseq <- gbn_rnaseq %>%
  left_join(data_genome, by = "gene")

MMC_rnaseq <- na.omit(MMC_rnaseq)
bldB_rnaseq <- na.omit(bldB_rnaseq)
ccr1_rnaseq <- na.omit(ccr1_rnaseq)
ethanol_sekurowa_sven <- na.omit(ethanol_sekurowa_sven)
gbn_rnaseq <- na.omit(gbn_rnaseq)
  
write.csv(bldB_rnaseq, "final_data/BldB_Avramova_2023_sven.csv", row.names = FALSE)
 write.csv(MMC_rnaseq, "final_data/MMC_Stratton_2022_sven.csv", row.names = FALSE)
 write.csv(ccr1_rnaseq, "final_data/ccr1_Lei_2024.csv", row.names = FALSE)
 write.csv(ethanol_sekurowa_sven, "final_data/ETOHschock_Sekurova_2022_sven.csv", row.names = FALSE)
 write.csv(gbn_rnaseq, "final_data/Gbn_Du_2022.csv", row.names = FALSE)





```

```{r}

szafran <- read.csv("final_data/data_szafran2019.csv")

nazwy_zmiennych_szafran <- unique(szafran$add_variable)

nazwy_zmiennych_szafran

szafran$add_variable <- gsub("M145PS04", "PS04M145", szafran$add_variable)
szafran$add_variable <- gsub("M145PS04novobiocyna", "PS04M145novobiocyna", szafran$add_variable)
szafran$add_variable <- gsub("M145PS04thio", "PS04M145thio", szafran$add_variable)

write.csv(szafran, "final_data/data_szafran2019.csv", row.names = FALSE)
```

```{r}
martyna <- read.csv("new_data/satkr_martyna.txt", sep = '\t')

martyna <- martyna %>% select(gene, FDR, foldodwrot, add_variable)

martyna$FDR <- gsub(",", ".", martyna$FDR)
martyna$foldodwrot <- gsub(",", ".", martyna$foldodwrot)

martyna$FDR <- as.numeric(martyna$FDR)
martyna$fold <- as.numeric(martyna$fold)
martyna$foldodwrot <- as.numeric(martyna$foldodwrot)

is.na(martyna$foldodwrot == NA)


martyna$logFC <- log2(martyna$foldodwrot)

martyna <- martyna %>% select(gene, FDR, logFC, add_variable)

martyna <- martyna %>%
  left_join(data_genome, by = "gene")


write.csv(martyna, "final_data/RNAseq_Martyna.csv", row.names = FALSE)

```

```{r}
martyna <- read.csv("final_data/RNAseq_Martyna.csv")

colnames(martyna)

nazwy_martyna <- unique(martyna$add_variable)
nazwy_martyna

#"PS04vsMGP10" "PS04vsMGP11" "PS04vsMGP12" "PS04vsMGP14" "PS04vsMGHM5" "M145vsMGM10" "M145vsMGM11" "M145vsMGM12" "M145vsMGM14"
#"MGP10vsPS04" "MGP11vsPS04" "MGP12vsPS04" "MGP14vsPS04" "MGHM5vsPS04" "MGM10vsM145" "MGM11vsM145" "MGM12vsM145" "MGM14vsM145"

martyna$add_variable <- gsub("PS04vsMGP10", "MGP10vsPS04", martyna$add_variable)
martyna$add_variable <- gsub("PS04vsMGP11", "MGP11vsPS04", martyna$add_variable)
martyna$add_variable <- gsub("PS04vsMGP12", "MGP12vsPS04", martyna$add_variable)
martyna$add_variable <- gsub("PS04vsMGP14", "MGP14vsPS04", martyna$add_variable)
martyna$add_variable <- gsub("PS04vsMGHM5", "MGHM5vsPS04", martyna$add_variable)
martyna$add_variable <- gsub("M145vsMGM10", "MGM10vsM145", martyna$add_variable)
martyna$add_variable <- gsub("M145vsMGM11", "MGM11vsM145", martyna$add_variable)
martyna$add_variable <- gsub("M145vsMGM12", "MGM12vsM145", martyna$add_variable)
martyna$add_variable <- gsub("M145vsMGM14", "MGM14vsM145", martyna$add_variable)



write.csv(martyna, "final_data/RNAseq_Martyna.csv", row.names = FALSE)

```
```{r}

create_empty_rnaseq_df <-   data.frame(
    gene = NA_character_,
    logFC = NA_real_,
    FDR = NA_real_,
    add_variable = NA_character_,
    start = NA_integer_,
    end = NA_integer_,
    width = NA_integer_,
    strand = NA_character_,
    data_name = NA_character_,
    stringsAsFactors = FALSE
  )

```

```{r}
 chipseq_smirdova <- read.csv("final_data/data_chip_seq/chipseq_smirdova.txt", sep = '\t')
  write.csv(chipseq_smirdova, "final_data/data_chip_seq/chipseq_smirdova.csv", row.names = FALSE)

  
  
```



```{r}
strzalka <- read.csv("final_data/data_chip_seq/hupAS_strzalka/data_hupA_chipseq_macs.txt", sep = ' ')

strzalka$chrom <- "Chromosome"
strzalka <- strzalka %>% select(chrom, start, end, rodzaj)
colnames(strzalka) <- c("chrom", "chromStart", "chromEnd", "name")

 write.csv(strzalka, "final_data/data_chip_seq/hupAS_strzalka_chipseq.csv", row.names = FALSE)

```
```{r}
strzalka <- read.csv("final_data/data_chip_seq/hupAS_strzalka_chipseq.csv")


```

```{r}

data_template <- read.csv("templates_for_app.txt", sep='\t')  

read_xlsx("datasets/templates_for_app.xlsx")

```

